### PR TITLE
Restrict tab focus to within modal dialogues when open. fix issues #1070 and #1072 

### PR DESCRIPTION
**Pull request by thattonBL**
Restrict tab focus to within modal dialogues when open

fix issue #1070
fix issue #1072

- Add key listener override on Dialogue.ts which is the base class for most Dialogues
- Added code to do the same thing for the React DownloadDialogue.tsx which serves as (yes you guessed it) the Download DIalogue

### CODE REVIEW

File: src/content-handlers/iiif/extensions/uv-openseadragon-extension/DownloadDialogue.tsx [hunk 1]
```diff
@@ -561,7 +602,7 @@ const DownloadDialogue = ({
           <ol className="options">
             {isDownloadOptionAvailable(DownloadOption.CURRENT_VIEW) && (
               <li className="option single">
-                <button
+                <button ref={firstFocusableElementRef}
```
Comment (demiankatz) [inline src/content-handlers/iiif/extensions/uv-openseadragon-extension/DownloadDialogue.tsx:605]: Because of the isDownloadOptionAvailable check around each button, there is no guarantee about which buttons will actually be rendered -- thus, I think we're going to need somewhat more sophisticated logic to figure out which will be the first and last focusable elements. Maybe we need to refactor this to build an array of buttons and then build the ol/li from the array, so we can apply the references in the appropriate places.

File: src/content-handlers/iiif/modules/uv-shared-module/Dialogue.ts [hunk 2]
```diff
@@ -28,7 +28,7 @@ export class Dialogue<
 
   create(): void {
     this.setConfig("dialogue");
-
+
```
Comment (demiankatz) [inline src/content-handlers/iiif/modules/uv-shared-module/Dialogue.ts:31]: Nitpick: unnecessary whitespace change.

**Review by demiankatz — CHANGES_REQUESTED** [review]
Thanks, @thattonBL -- looks like a good start, but I think I spotted one problem (and one trivial thing you're free to ignore if you wish, but which might clean up the diffs slightly). :-)

File: src/content-handlers/iiif/modules/uv-shared-module/Dialogue.ts (hunk 2 above)
Comment (thattonBL) [reply src/content-handlers/iiif/modules/uv-shared-module/Dialogue.ts]: Hi @demiankatz I have removed this now

**Comment by demiankatz** [comment]
Superseded by #1163.
